### ISSUE TITLE
🐞 fix(sampling): KCenterGreedy returns 1 sample when coreset_size == 0

### ISSUE THREAD

**Issue by shaun0927** [open-edge-platform/anomalib#3544]
## Describe the bug

`KCenterGreedy.select_coreset_idxs()` violates its size contract when
`int(n_observations * sampling_ratio) == 0`. The caller requests zero samples
but receives one.

PR #3509 added the random seed index to the returned coreset and paired it
with `range(self.coreset_size - 1)`. When `coreset_size` is 0 the range
becomes `range(-1)` (empty), so the seed index is appended and never popped.
The returned list has length 1 instead of 0.

## Steps to reproduce

```python
import torch
from anomalib.models.components.sampling import KCenterGreedy

sampler = KCenterGreedy(embedding=torch.randn(10, 4), sampling_ratio=0.01)
idxs = sampler.select_coreset_idxs()

assert sampler.coreset_size == 0     # passes
assert len(idxs) == 0                 # fails — len(idxs) == 1
```

`sampling_ratio=0.01` together with a 10-row embedding (or any small ratio /
small embedding combination) triggers the path.

## Expected behavior

When `coreset_size == 0`, return an empty list. Silently seeding with a
single observation distorts the downstream distance metric for callers that
rely on the documented size contract.

## OS information

- Python: 3.10+
- anomalib: `main` @ 91d16ef0
- Torch: 2.6+

## Additional context

This is a direct regression introduced by PR #3509. Fix is a single-line
guard; PR with regression test ready to submit:

```python
if self.coreset_size == 0:
    return []
```
